### PR TITLE
RM-7971 Fix race condition in AsyncLocalStorageProviderTest

### PR DESCRIPTION
**Pull request by a-ctor**
https://re-motion.atlassian.net/browse/RM-7971

### CODE REVIEW

File: Remotion/Core/UnitTests/Context/AsyncLocalStorageProviderTest.cs [hunk 1]
```diff
@@ -102,42 +103,37 @@ public void SetData_FromOtherThread_DoesNotOverride ()
...
+              else
+              {
+                result = _provider.GetData (c_testKey);
+                completed = true;
+              }
+
+              Thread.MemoryBarrier();
               resetEvent.Set();
             });
         resetEvent.WaitOne();
```
Comment (MichaelKetting) [inline Remotion/Core/UnitTests/Context/AsyncLocalStorageProviderTest.cs:133]: Say, should we add a timeout to the WaitOne and fail the test if it's reached so the code can't hang the build if there's a thread-starvation bug or just environment condition? And if if so, new ticket (please create and schedule) or right here?

**Review by MichaelKetting — APPROVED** [review]
Looks good, please check comment.

File: Remotion/Core/UnitTests/Context/AsyncLocalStorageProviderTest.cs (hunk 1 above)
Comment (a-ctor) [reply Remotion/Core/UnitTests/Context/AsyncLocalStorageProviderTest.cs]: I don't think that such a case will ever come up but I've added a timeout of 3sec onto the wait.

File: Remotion/Core/UnitTests/Context/AsyncLocalStorageProviderTest.cs [hunk 2]
```diff
@@ -102,42 +103,39 @@ public void SetData_FromOtherThread_DoesNotOverride ()
...
               resetEvent.Set();
             });
-        resetEvent.WaitOne();
-      }
-      finally
-      {
-        ThreadPool.SetMaxThreads (maxWorkerThreads, maxCompletionPortThreads);
-        ThreadPool.SetMinThreads (minWorkerThreads, minCompletionPortThreads);
+
+        if (!resetEvent.WaitOne (3 * 1000))
```
Comment (MichaelKetting) [inline Remotion/Core/UnitTests/Context/AsyncLocalStorageProviderTest.cs:134]:
```suggestion
        if (!resetEvent.WaitOne (TimeSpan.FromSeconds (3)))
```

Yes, I know, highly unlikely. But a hanging build on the build server trumps all the 'unlikelies' we can see and easily prevent. The problem is, first you need to notice, then you need to be able to attach a debugger before someone kills it, and then you have a chance to figure it out. Unfortunately, the in-general scenario of the hanging build without obivous cause isn't just theory, so going the extra inch is usually worth the effort. Especially if the lead reviewer is usually the one to find the hanging build early in the morning and then gets to have some fun.

**Review by MichaelKetting — APPROVED** [review]
Please update the syntax and we're done.

**Pull request merged**